src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -47,67 +45,56 @@ import {
 import { serializeTxIndex } from 'state/slices/txHistorySlice/utils'
 import { useAppSelector } from 'state/store'
 
-import { getSwapperManager } from '../hooks/useSwapper/swapperManager'
 import type { TS } from '../types'
 import { TradeRoutePaths } from '../types'
 import { WithBackButton } from '../WithBackButton'
 import { AssetToAsset } from './AssetToAsset'
 import { ReceiveSummary } from './ReceiveSummary'
 
-type TradeConfirmParams = {
-  fiatRate: string
-}
-
 export const TradeConfirm = ({ history }: RouterProps) => {
   const borderColor = useColorModeValue('gray.100', 'gray.750')
   const [sellTxid, setSellTxid] = useState('')
   const [buyTxid, setBuyTxid] = useState('')
-  const flags = useSelector(selectFeatureFlags)
-  const [swapperName, setSwapperName] = useState<string>('')
   const [executedTradeAmountConstants, setExecutedTradeAmountConstants] =
     useState<ReturnType<typeof getTradeAmountConstants>>()
+  const { bestTradeSwapper } = useSwapper()
```

Comment:
does this get set when we setup the quote and not change after executing the quote? this existed because we had a bug where the entire swapper would rerender on the confirm model, switch back to the default pair, and show the best swapper (currently 0x) for that pair instead of the trade executed pair.

Reply:
You're absolutely right, I shouldn't have removed your `useEffect`. Restored in https://github.com/shapeshift/web/pull/3084/commits/5e3d528c9e05126bb2268e628aacb959ed84e18f.

`trade` stabilised in https://github.com/shapeshift/web/pull/3084/commits/4c9752ccca3d178e806959908854f387e4f783c9.